### PR TITLE
chore: add preprocessing function to convert tools file

### PR DESCRIPTION
**Pull request by Yuan325**
Add preprocessing function during parsing of tools file to convert v2 tools file.

Related #817

Future updates will include: 
* Update unmarshal function for ToolsFile to convert config file (test will fail since the yaml tag is not yet updated).
* Update yaml tag (test will pass).

### CODE REVIEW

File: cmd/test.db
Comment (averikitsch) [inline cmd/test.db:1]: Is this file required?

File: cmd/root.go [hunk 1]
```diff
@@ -425,6 +426,106 @@ func parseEnv(input string) (string, error) {
...
+	resourceOrder := []string{}
+	lookup := make(map[string]yaml.MapSlice)
+	for _, item := range input {
+		key, ok := item.Key.(string)
+		if !ok {
+			return nil, fmt.Errorf("unexpected non-string key in input: %v", item.Key)
+		}
+		if slice, ok := item.Value.(yaml.MapSlice); ok {
+			// convert authSources to authServices
+			if key == "authSources" {
```
Comment (averikitsch) [inline cmd/root.go]: Do we need to continue to convert authSources? Should we throw a deprecation error?

File: cmd/root.go [hunk 2]
```diff
@@ -425,6 +426,106 @@ func parseEnv(input string) (string, error) {
...
+			// works even if lookup[key] is nil
+			lookup[key] = append(lookup[key], slice...)
+			// preserving the resource's order of original toolsFile
+			if !slices.Contains(resourceOrder, key) {
+				resourceOrder = append(resourceOrder, key)
+			}
+		} else {
+			// toolsfile is already v2
+			if key == "kind" {
+				return raw, nil
```
Comment (averikitsch) [inline cmd/root.go]: Is there any chance we would return early but need to check more keys?

File: cmd/test.db
Comment (Yuan325) [reply cmd/test.db:1]: removed.

File: cmd/root.go (hunk 1 above)
Comment (Yuan325) [reply cmd/root.go]: @averikitsch do we want to stop support backwards compatible for this?

File: cmd/root.go (hunk 2 above)
Comment (Yuan325) [reply cmd/root.go]: Only if user combine both v1 and v2 formatting in the same config file. Do we want to support that? (I feel like it will get messy) If user have one file in the v1 format, and another in v2, this will still work.

File: cmd/root.go (hunk 1 above)
Comment (Yuan325) [reply cmd/root.go]: keeping for backwards compatibility.

File: cmd/root.go (hunk 2 above)
Comment (averikitsch) [reply cmd/root.go]: I see this as a potential situation. I think we should ensure that this edge case is handled.

Comment (Yuan325) [reply cmd/root.go]: Updated, and added test case for that.

File: cmd/root.go [hunk 3]
```diff
@@ -425,6 +426,129 @@ func parseEnv(input string) (string, error) {
...
+			return nil, err
+		}
+		for _, item := range input {
+			key, ok := item.Key.(string)
+			if !ok {
+				return nil, fmt.Errorf("unexpected non-string key in input: %v", item.Key)
+			}
+			// check if the key is config file v1's key
+			if slices.Contains(v1keys, key) {
+				// check if value convertion to yaml.MapSlice successfully
```
Comment (averikitsch) [inline cmd/root.go]:
```suggestion
				// check if value conversion to yaml.MapSlice successfully
```
